### DISCUSSION TITLE
Unable to read output events from MobiFlight

### DISCUSSION THREAD

**Discussion by markmalek** [nguyenquyhy/Flight-Tracker-StreamDeck#350] [Q&A]
I've been attempting to read events from MobiFlight (output) but Flight Tracker seems to be unable to read these events.

For example: `FNX320 Electrical Battery 1 Button Annunciator Upper`  or `MOBIFLIGHT_FNX320 Electrical Battery 1 Button Annunciator Upper`

I can read this event in MobiFlight but not Flight Tracker. I've connected VSCode debugger to my stream deck and can see that this event (and any like it from MobiFlight) are ignored and never register.

Is this not currently supported? If not, I've been poking around the code to see where I can add support for these types of events but unable to find exactly where Flight Tracker reads and parses event names.

**Comment by matslarson** [comment]
I don’t know if this project it still maintained but I stumbled on it this week and ran into the same issue. The MobiFlight integration (which is fantastic @nguyenquyhy!) let’s me _send_ virtually any event but I’m still limited to setting feedback only on the predefined built-ins 😔

None of the obvious methods worked yesterday but I’ll poke around the code and see if it’s something silly I’m missing. My hunch is it just doesn’t support reading arbitrary values from WASM

**Reply by nguyenquyhy** [reply]
It is still maintained. And I still have a lot of new features planned. I am just too busy at work in the recent months.

Regarding reading variables from Mobiflight, it is not supported yet. There is a fork that has already implemented the support, but I want the implementation to be bit more robust, so I have something a bit different in mind. It will come some time this year, or maybe early next year.

**Comment by nguyenquyhy** [comment]
What you mentioned is not events but custom data sent from the sim via Client Data. This is not supported yet in this repo. One of the fork already has the support implemented, but I have something a bit different in mind to make it more robust.

I can't say when it will come yet as I am too busy at work recently. But I'll aim for end of this year or early next year.
